packages/mwp-app-render/src/components/WriteStyles.jsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Helmet from 'react-helmet';
+
+/**
+ * @module WriteStyles
+ *
+ * Adds style module CSS to `head` for both
+ * server and client render via `Helmet`.
+ *
+ * Usage:
+ * ```
+ * import WriteStyles from 'mwp-app-render/lib/components/WriteStyles';
+ * import styles from './myComponent.module.scss';
+ *
+ * const MyComponent = (props) => {
+ *		return (
+ *			<div>
+	 *			<WriteStyles styles={styles} />
+	 *			<p className={styles.someClass}>Hello world!</p>
+ *			</div>
+ *		);
+ * };
+ * ```
+ */
+const WriteStyles = props => {
```

Comment:
`WriteStyles` is a fine name, but I'm wondering if it's too implementation-specific - i.e. it refers to the fact that in its current form, the styles are written into the markup, but from the developers perspective, they just want to know that the styles are being made available, not _how_ they are being made available. The difference might be academic, but what about making `WriteStyles` a wrapper that implicitly provides a 'styled context' to its children, either as a HOC function (e.g. `withStyles(styles)(WrappedComponent)`) or as a component that simply renders its children

```jsx
<StyleWrapper styles={styles}>
  <WrappedComponent />
</StyleWrapper>
```

I would actually lean toward the latter because it matches my intuition about how I would want styles to be provided to my component (even if it doesn't actually match what the implementation does), and it might give you a good excuse to play with [React Fragments](https://reactjs.org/docs/fragments.html#short-syntax) like

```jsx
class StyleWrapper extends React.Component {
  render() {
    <>
      <Helmet>...</Helmet>
      {children}
    </>
  }
}
```

If you think `WriteStyles` as a sibling element is the right way to set this up, that's cool too, no change necessary

Reply:
I was also thinking `StyleProvider` would be a nice name, but `Provider` has become the convention for labeling React Context providers, and we don't want to make that implication here

Reply:
I had actually named this `StylesProvider`, and @mperrotti pointed out that it's not _actually_ providing anything in the way that other `[Thing]Provider` components in our apps do.

I've been divided on sibling vs. wrapping with `WriteStyles`; using it as a sibling seems to break a convention, so I think i'm going to change this to wrap children. Perhaps `<StyleWith styles={styles}>...children...</StyleWith>`?

Reply:
sounds good to me